rust/connlib/tunnel/src/io.rs

```diff
@@ -183,6 +182,14 @@ impl Io {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum DnsQueryError {
```

Comment:
And these are the only two reasons why we'd reject a query without even attempting to contact an upstream server: Either we have too many queries buffered, or the packet is not destined for one of our upstreams somehow.

Is that really "No upstream DNS server configured" and not "DNS query is not for any of our servers"?